### ISSUE TITLE
Updating pre master->main instructions

### ISSUE THREAD

**Issue by cgreene** [manubot/rootstock#411]
I was just updating the rootstock for the COVID-19 review and realized that when I pulled from rootstock master as described in the setup instructions on the repository, it told me I was already up to date. Once I realized that things had moved to `main` and I pulled from rootstock main, things worked.

Is there a plan to help folks make this shift? Otherwise, it seems folks might be stuck with an old rootstock.

**Comment by dhimmel** [comment]
> as described in the setup instructions on the repository

I see, so your were looking at the setup instructions for the covid manuscript and not rootstock: https://github.com/greenelab/covid19-review/blob/3564964529462913b8e9eb1a153f2e830073267f/SETUP.md#merging-upstream-rootstock-changes

And not the [current setup instruction](https://github.com/manubot/rootstock/blob/87c11957cf69b73ebed5518f7928b99adb1dc5d6/SETUP.md#merging-upstream-rootstock-changes) on Rootstock.

We could consider:

1. deleting the rootstock master branch (will have to test what the effect of this will be)
2. adding a commit to master that updates the SETUP instructions
3. detecting manuscripts still using master and notify them

**Comment by cgreene** [comment]
Exactly! I was kind of imagining 2, but it would need to merge cleanly if someone then pulled from main.

**Comment by agitter** [comment]
This issue and the recent https://github.com/manubot/rootstock/issues/407#issuecomment-766370375 suggests the branch transition wasn't as clean as we had hoped.  If we delete the master branch (option 1), won't that create 404 errors?  I don't think GitHub redirects automatically like it does when renaming or transferring a repository.  We could test this in a different repo.

Option 2 could be a reasonable workaround.

Option 3 would affect a lot of manuscripts.  I don't think we should ping users for an issue that isn't breaking their existing builds.

As an option 4, I also considered a GitHub Actions workflow that would keep main and master in sync.  That would keep any links to master branch rootstock resources current.  However, we wouldn't want that workflow to appear in manuscripts derived from rootstock.

**Comment by dhimmel** [comment]
> I don't think GitHub redirects automatically like it does when renaming or transferring a repository.

[Links to deleted branches now redirect to the default branch](https://github.blog/changelog/2020-07-17-links-to-deleted-branches-now-redirect-to-the-default-branch/) as of July 17, 2020. This "only affects view links".

But I don't think browser URLs are our main concern. Instead we need to know what happens to a repo using `master` if they run `git pull --no-ff --no-rebase --no-commit rootstock master` and we've deleted master.

> keep main and master in sync

This is an interesting idea. I think the current code in `main` still supports a default `master` such that this might just work.

**Comment by dhimmel** [comment]
> adding a commit to master that updates the SETUP instructions

Perhaps the easiest an most immediate solution would be to fast forward master by one commit to 690aac113b4ee7dbb32a0af10a3d36c51a207025. Then users with old upgrade instructions would at least be able to get the newer upgrade instructions. It also would leave the most recent master commit in a state that is somewhat explanatory of why the repo has a master branch (for back compat).

Any objections to this? It wouldn't rule out the additional step of "keeping main and master in sync"

**Comment by agitter** [comment]
> Perhaps the easiest an most immediate solution would be to fast forward master by one commit to 690aac1

No objections here.  That makes sense as an initial step.

> Links to deleted branches now redirect to the default branch

Nice.  I didn't know about that behavior.  It doesn't solve all of our problems but is a step in the right direction.

**Comment by dhimmel** [comment]
> That makes sense as an initial step.

Done. CI logs at https://github.com/manubot/rootstock/runs/1916094763
